### PR TITLE
nest: init at 3.3

### PR DESCRIPTION
**Pull request by jiegec**
###### Description of changes

NEST is a simulator for spiking neural network models that focuses on the dynamics, size and structure of neural systems rather than on the exact morphology of individual neurons. 

Homepage: https://www.nest-simulator.org/

###### Things done

- Built on platform(s)
  - [x] x86_64-linux
  - [ ] aarch64-linux
  - [ ] x86_64-darwin
  - [x] aarch64-darwin
- [ ] For non-Linux: Is `sandbox = true` set in `nix.conf`? (See [Nix manual](https://nixos.org/manual/nix/stable/command-ref/conf-file.html))
- [x] Tested, as applicable:
  - [NixOS test(s)](https://nixos.org/manual/nixos/unstable/index.html#sec-nixos-tests) (look inside [nixos/tests](https://github.com/NixOS/nixpkgs/blob/master/nixos/tests))
  - and/or [package tests](https://nixos.org/manual/nixpkgs/unstable/#sec-package-tests)
  - or, for functions and "core" functionality, tests in [lib/tests](https://github.com/NixOS/nixpkgs/blob/master/lib/tests) or [pkgs/test](https://github.com/NixOS/nixpkgs/blob/master/pkgs/test)
  - made sure NixOS tests are [linked](https://nixos.org/manual/nixpkgs/unstable/#ssec-nixos-tests-linking) to the relevant packages
- [x] Tested compilation of all packages that depend on this change using `nix-shell -p nixpkgs-review --run "nixpkgs-review rev HEAD"`. Note: all changes have to be committed, also see [nixpkgs-review usage](https://github.com/Mic92/nixpkgs-review#usage)
- [x] Tested basic functionality of all binary files (usually in `./result/bin/`)
- [22.11 Release Notes (or backporting 22.05 Release notes)](https://github.com/NixOS/nixpkgs/blob/master/CONTRIBUTING.md#generating-2211-release-notes)
  - [ ] (Package updates) Added a release notes entry if the change is major or breaking
  - [ ] (Module updates) Added a release notes entry if the change is significant
  - [ ] (Module addition) Added a release notes entry if adding a new NixOS module
  - [ ] (Release notes changes) Ran `nixos/doc/manual/md-to-db.sh` to update generated release notes
- [x] Fits [CONTRIBUTING.md](https://github.com/NixOS/nixpkgs/blob/master/CONTRIBUTING.md).

### CODE REVIEW

**Comment by jiegec** [comment]
Result of `nixpkgs-review pr 198872` run on x86_64-linux [1](https://github.com/Mic92/nixpkgs-review)
<details>
  <summary>6 packages built:</summary>
  <ul>
    <li>nest</li>
    <li>nest-mpi</li>
    <li>python310Packages.nest</li>
    <li>python310Packages.nest-mpi</li>
    <li>python39Packages.nest</li>
    <li>python39Packages.nest-mpi</li>
  </ul>
</details>

**Comment by jiegec** [comment]
Result of `nixpkgs-review pr 198872` run on aarch64-darwin [1](https://github.com/Mic92/nixpkgs-review)
<details>
  <summary>6 packages built:</summary>
  <ul>
    <li>nest</li>
    <li>nest-mpi</li>
    <li>python310Packages.nest</li>
    <li>python310Packages.nest-mpi</li>
    <li>python39Packages.nest</li>
    <li>python39Packages.nest-mpi</li>
  </ul>
</details>

**Comment by jiegec** [comment]
Result of `nixpkgs-review pr 198872` run on x86_64-linux [1](https://github.com/Mic92/nixpkgs-review)
<details>
  <summary>6 packages built:</summary>
  <ul>
    <li>nest</li>
    <li>nest-mpi</li>
    <li>python310Packages.nest</li>
    <li>python310Packages.nest-mpi</li>
    <li>python39Packages.nest</li>
    <li>python39Packages.nest-mpi</li>
  </ul>
</details>

File: pkgs/top-level/python-packages.nix [hunk 1]
```diff
@@ -6049,6 +6049,10 @@ self: super: with self; {
 
   nessclient = callPackage ../development/python-modules/nessclient { };
 
+  nest = toPythonModule(pkgs.nest.override { withPython = true; python3 = python; });
+
+  nest-mpi = toPythonModule(pkgs.nest-mpi.override { withPython = true; python3 = python; });
```
Comment (SuperSandro2000) [inline pkgs/top-level/python-packages.nix:6054]: We should only include one. How about nest but with mpi?

**Review by SuperSandro2000 — CHANGES_REQUESTED** [review]

File: pkgs/top-level/python-packages.nix (hunk 1 above)
Comment (jiegec) [reply pkgs/top-level/python-packages.nix]: Nice catch. Done in the new commit.

**Comment by SuperSandro2000** [comment]
@ofborg eval

**Pull request merged**